packages/core/src/models/branding-api-response.ts

```diff
@@ -0,0 +1,590 @@
+   */
+  importURL?: string;
+}
+
+/**
+ * Interface for the Login Box Preferences.
+ */
+export interface BrandingPreferenceButtonsInterface {
+  /**
+   * Social, External IDP Connection Button Preference.
+   */
+  externalConnection: ElementStateInterface<ButtonStyleAttributesInterface>;
+  /**
+   * Primary Button Preferences.
+   */
+  primary: ElementStateInterface<Omit<ButtonStyleAttributesInterface, 'background'>>;
+  /**
+   * Secondary Button Preferences.
+   */
+  secondary: ElementStateInterface<Omit<ButtonStyleAttributesInterface, 'background'>>;
+}
+
+/**
+ * Interface for the Login Box Preferences.
+ */
+export interface BrandingPreferenceInputInterface {
+  /**
+   * Input field background.
+   */
+  background: BackgroundStyleAttributesInterface;
+  /**
+   * Secondary Button Preferences.
+   */
+  border: Pick<BorderStyleAttributesInterface, 'borderRadius' | 'borderColor'>;
+  /**
+   * Input Field Font Preferences.
+   */
+  font: FontStyleAttributesInterface;
+  /**
+   * Input Labels Preferences.
+   */
+  labels: {
+    /**
+     * Input Labels Font Preferences.
+     */
+    font: FontStyleAttributesInterface;
+  };
+}
+
+export interface BrandingPreferenceLoginBoxInterface {
+  /**
+   * Login Box Background.
+   */
+  background: BackgroundStyleAttributesInterface;
+  /**
+   * Login Box Border.
+   */
+  border: BorderStyleAttributesInterface;
+  /**
+   * Login Box Font.
+   */
+  font: FontStyleAttributesInterface;
+}
+
+/**
+ * Interface Branding preference layout.
+ */
+export type BrandingPreferenceLayoutInterface = StrictBrandingPreferenceLayoutInterface &
+  Partial<DynamicBrandingPreferenceLayoutInterface>;
+
+/**
+ * Strict Interface Branding preference layout.
+ */
+export interface StrictBrandingPreferenceLayoutInterface {
+  /**
+   * The active layout.
+   */
+  activeLayout: PredefinedLayouts;
+}
+
+/**
+ * Interface dynamic branding preference layout.
+ */
+export type DynamicBrandingPreferenceLayoutInterface = BrandingPreferenceSideImageLayoutInterface &
+  BrandingPreferenceSideAlignedLayoutInterface;
+
+/**
+ * Left Image and Right Image layouts preference interface.
+ */
+export interface BrandingPreferenceSideImageLayoutInterface {
+  sideImg: BrandingPreferenceImageInterface;
+}
+
+/**
+ * Left Aligned and Right Aligned layouts preference interface.
+ */
+export interface BrandingPreferenceSideAlignedLayoutInterface {
+  productTagLine: string;
+}
+
+/**
+ * Interface Branding preference configurations.
+ */
+export interface BrandingPreferenceConfigInterface {
+  /**
+   * Should the changes be published?
+   */
+  isBrandingEnabled: boolean;
+  /**
+   * Should remove default branding.
+   * @deprecated Renamed to `removeDefaultBranding` to keep it common.
+   */
+  removeAsgardeoBranding?: boolean;
+  /**
+   * Should remove default branding.
+   */
+  removeDefaultBranding?: boolean;
+}
+
+/**
+ * Enum for Branding Preference Types.
+ */
+export enum BrandingPreferenceTypes {
+  App = 'APP',
```

Comment:
Let's add comments to these as well for consistency

Reply:
added with 4a579777913a1e266bb02d9535a6176ce2791222